File: samples/samples-outofproc/TriggerBindingSamples/ProductsTrigger.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

using System.Collections.Generic;
using Microsoft.Azure.Functions.Worker;
using Microsoft.Azure.Functions.Worker.Extensions.Sql;
using Microsoft.Azure.WebJobs.Extensions.Sql.SamplesOutOfProc.Common;
using Microsoft.Extensions.Logging;
using Newtonsoft.Json;
using System;

namespace Microsoft.Azure.WebJobs.Extensions.Sql.SamplesOutOfProc.TriggerBindingSamples
{
    public class ProductsTrigger
    {
        private static readonly Action<ILogger, string, Exception> _loggerMessage = LoggerMessage.Define<string>(LogLevel.Information, eventId: new EventId(0, "INFO"), formatString: "{Message}");

        [Function("ProductsTrigger")]
        public static void Run(
            [SqlTrigger("[dbo].[Products]", "SqlConnectionString")]
            IReadOnlyList<SqlChange<Product>> changes, FunctionContext context)
        {
            foreach (SqlChange<Product> change in changes)
            {
                // The output is used to inspect the trigger binding parameter in test methods.
                _loggerMessage(context.GetLogger("ProductsTrigger"), "SQL Change: " + JsonConvert.SerializeObject(change), null);
            }
        }
    }
}

File: samples/samples-java/src/main/java/com/function/ProductsTrigger.java
/**
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 */

package com.function;

import com.microsoft.azure.functions.ExecutionContext;
import com.microsoft.azure.functions.annotation.FunctionName;
import com.microsoft.azure.functions.sql.annotation.SQLTrigger;
import com.function.Common.SqlChangeProduct;
import com.google.gson.Gson;

import java.util.logging.*;

public class ProductsTrigger {
    @FunctionName("ProductsTrigger")
    public void run(
            @SQLTrigger(
                name = "changes",
                tableName = "[dbo].[Products]",
                connectionStringSetting = "SqlConnectionString")
                SqlChangeProduct[] changes,
            ExecutionContext context) {
        Gson gson = new Gson();
        Logger logger = context.getLogger();
        logger.log(Level.INFO, "SQL Changes: " + changes.length);
        for (SqlChangeProduct change : changes) {
            // The output is used to inspect the trigger binding parameter in test methods.
            logger.log(Level.INFO, "SQL Change: " + gson.toJson(change));
        }
    }
}